### ISSUE TITLE
[🐛]🔥 Previously healthy code creates a build that crashes on startup (logs point to Firebase connection/auth issue)

### ISSUE THREAD

**Issue by jakechao-zeal** [invertase/react-native-firebase#7697]
## Issue

I am using Expo Managed Workflow (Expo SDK 49) to create an IOS build for my app that uses React Native Firebase (alongside React Redux Firebase). I am posting here since I believe the crash is being caused by some connection to Firebase which RNFB might be handling. If the crash logs indicate that this is not related to RNFB, I apologize for posting the issue in the wrong place (any guidance would still be appreciated). 

When trying to create a new build this week, I found that code, which was producing a healthy build last week, was now producing a build that crashes as soon as the app starts (right as soon as the splash appears). I uploaded the crashing build to TestFlight and got the crash logs, which I've posted below.

_What's most odd to me is that I changed no dependencies since last week when I was able to create a healthy build using this same code branch, package.json, and package-lock. Old builds before this week with the same code and dependencies are also still working - only new ones are failing. I'm relatively new to app development so if there's any guidance on what may have changed such that the builds are now crashing, that would be much appreciated._ 

What I've tried with no luck includes:

- Updating packages to RNFB 18.9.0 
- Updating packages to RNFB 19.0.1
- Ensuring that packages are all aligned with the Expo SDK I am on 

The crash still occurs on app startup even after update. 

**Note: I am on RNFB 18.5.0 and know that there have been issues with Expo and RNFB versions lower than 19.0.1, but I've followed the guidance in #7666 to adjust my pods so that the IOS Firebase SDK 10.22.0 is used.** 

IOS Crash Logs from Test Flight (included only the Thread that Crashed for brevity) - replaced AppName for our app's name in the crash logs

```
---------------------------------------
Desymbolication will use: 
---------------------------------------
Crash stack
0   libsystem_kernel.dylib        	0x00000001d7479fbc __pthread_kill + 8
1   libsystem_pthread.dylib       	0x00000001f9e17680 pthread_kill + 268 (pthread.c:1681)
2   libsystem_c.dylib             	0x0000000197d92b90 abort + 180 (abort.c:118)
3   libsystem_c.dylib             	0x0000000197d91e80 __assert_rtn + 284 (assert.c:94)
4   AppName                   	0x0000000102abfdf0 GRPC_ec_felem_neg.cold.1 + 40 (felem.c:66)
5   AppName                   	0x0000000102a069c0 GRPC_ec_felem_neg + 148 (felem.c:66)
6   AppName                   	0x00000001029f823c ec_group_set_a_minus3 + 56 (ec.c:106)
7   AppName                   	0x00000001029f6f34 GRPC_EC_group_p256_do_init + 252 (ec.c:171)
8   AppName                   	0x00000001029f6f34 GRPC_EC_group_p256_init + 264 (ec.c:144)
9   libsystem_pthread.dylib       	0x00000001f9e153d4 __pthread_once_handler + 76 (pthread.c:1935)
10  libsystem_platform.dylib      	0x00000001f9d68eb4 _os_once_callout + 32 (lock.c:1180)
11  libsystem_pthread.dylib       	0x00000001f9e157a8 pthread_once + 100 (pthread.c:1945)
12  AppName                   	0x0000000102a465f0 GRPC_CRYPTO_once + 12 (thread_pthread.c:61)
13  AppName                   	0x00000001029f7348 GRPC_EC_group_p256 + 20 (ec.c:144)
14  AppName                   	0x00000001029f7348 GRPC_EC_GROUP_new_by_curve_name + 168 (ec.c:336)
15  AppName                   	0x00000001029f8fb8 GRPC_EC_KEY_new_by_curve_name + 40 (ec_key.c:142)
16  AppName                   	0x00000001028e782c populate_ssl_context(ssl_ctx_st*, tsi_ssl_pem_key_cert_pair const*, char const*) + 408 (ssl_transport_security.cc:843)
17  AppName                   	0x00000001028e702c tsi_create_ssl_client_handshaker_factory_with_options(tsi_ssl_client_handshaker_options const*, tsi_ssl_client_handshaker_factory**) + 348 (ssl_transport_security.cc:2162)
18  AppName                   	0x00000001028e3d88 grpc_ssl_credentials::InitializeClientHandshakerFactory(grpc_ssl_config const*, char const*, tsi_ssl_root_certs_store const*, tsi_ssl_session_cache*, tsi_ssl_client_handshaker_factory**) + 208 (ssl_credentials.cc:216)
19  AppName                   	0x00000001028e3bd0 grpc_ssl_credentials::grpc_ssl_credentials(char const*, grpc_ssl_pem_key_cert_pair*, grpc_ssl_verify_peer_options const*) + 172 (ssl_credentials.cc:68)
20  AppName                   	0x00000001028e42f4 grpc_ssl_credentials::grpc_ssl_credentials(char const*, grpc_ssl_pem_key_cert_pair*, grpc_ssl_verify_peer_options const*) + 16 (ssl_credentials.cc:49)
21  AppName                   	0x00000001028e42f4 grpc_ssl_credentials_create + 84 (ssl_credentials.cc:240)
22  AppName                   	0x00000001029b3594 grpc::SslCredentials(grpc::SslCredentialsOptions const&) + 176 (secure_credentials.cc:134)
23  AppName                   	0x0000000102361020 firebase::firestore::remote::(anonymous namespace)::CreateSslCredentials(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) + 60 (grpc_connection.cc:75)
24  AppName                   	0x0000000102360bc0 firebase::firestore::remote::GrpcConnection::CreateChannel() const + 356 (grpc_connection.cc:313)
25  AppName                   	0x000000010236096c firebase::firestore::remote::GrpcConnection::EnsureActiveStub() + 128 (grpc_connection.cc:295)
26  AppName                   	0x00000001023610b0 firebase::firestore::remote::GrpcConnection::CreateStream(absl::lts_20240116::string_view, firebase::firestore::credentials::AuthToken const&, std::__1::basic_string<char, std::__1::char_traits<cha... + 60 (grpc_connection.cc:341)
27  AppName                   	0x00000001023fe604 firebase::firestore::remote::Stream::ResumeStartWithCredentials(firebase::firestore::util::StatusOr<firebase::firestore::credentials::AuthToken> const&, std::__1::basic_string<char, std::__1::char_... + 132 (stream.cc:179)
28  AppName                   	0x00000001023ffcc4 firebase::firestore::remote::Stream::RequestCredentials()::$_0::operator()(absl::lts_20240116::optional<firebase::firestore::util::StatusOr<firebase::firestore::credentials::AuthToken> > const&, ab... + 56 (stream.cc:154)
29  AppName                   	0x00000001023ffcc4 decltype(std::declval<firebase::firestore::remote::Stream::RequestCredentials()::$_0::operator()(absl::lts_20240116::optional<firebase::firestore::util::StatusOr<firebase::firestore::credentials::A... + 56 (invoke.h:394)
30  AppName                   	0x00000001023ffcc4 void std::__1::__invoke_void_return_wrapper<void, true>::__call<firebase::firestore::remote::Stream::RequestCredentials()::$_0::operator()(absl::lts_20240116::optional<firebase::firestore::util::St... + 56 (invoke.h:479)
31  AppName                   	0x00000001023ffcc4 std::__1::__function::__alloc_func<firebase::firestore::remote::Stream::RequestCredentials()::$_0::operator()(absl::lts_20240116::optional<firebase::firestore::util::StatusOr<firebase::firestore::c... + 56 (function.h:185)
32  AppName                   	0x00000001023ffcc4 std::__1::__function::__func<firebase::firestore::remote::Stream::RequestCredentials()::$_0::operator()(absl::lts_20240116::optional<firebase::firestore::util::StatusOr<firebase::firestore::credent... + 72 (function.h:359)
33  AppName                   	0x00000001022ee6a8 std::__1::__function::__value_func<void ()>::operator()[abi:v15006]() const + 20 (function.h:512)
34  AppName                   	0x00000001022ee6a8 std::__1::function<void ()>::operator()() const + 20 (function.h:1197)
35  AppName                   	0x00000001022ee6a8 firebase::firestore::util::AsyncQueue::ExecuteBlocking(std::__1::function<void ()> const&) + 68 (async_queue.cc:89)
36  AppName                   	0x0000000102414c6c std::__1::__function::__value_func<void ()>::operator()[abi:v15006]() const + 20 (function.h:512)
37  AppName                   	0x0000000102414c6c std::__1::function<void ()>::operator()() const + 20 (function.h:1197)
38  AppName                   	0x0000000102414c6c firebase::firestore::util::Task::ExecuteAndRelease() + 184 (task.cc:102)
39  libdispatch.dylib             	0x0000000197cda300 _dispatch_client_callout + 20 (object.m:561)
40  libdispatch.dylib             	0x0000000197ce1894 _dispatch_lane_serial_drain + 748 (queue.c:3885)
41  libdispatch.dylib             	0x0000000197ce23c4 _dispatch_lane_invoke + 380 (queue.c:3976)
42  libdispatch.dylib             	0x0000000197ced004 _dispatch_root_queue_drain_deferred_wlh + 288 (queue.c:6913)
43  libdispatch.dylib             	0x0000000197cec878 _dispatch_workloop_worker_thread + 404 (queue.c:6507)
44  libsystem_pthread.dylib       	0x00000001f9e13964 _pthread_wqthread + 288 (pthread.c:2629)
45  libsystem_pthread.dylib       	0x00000001f9e13a04 start_wqthread + 8
```

---

### Javascript

<details><summary>Click To Expand</summary>
<p>

#### `package.json`:

```json
{
  "dependencies": {
    "@amplitude/analytics-react-native": "^1.4.5",
    "@cometchat-pro/react-native-chat": "^3.0.11",
    "@config-plugins/react-native-blob-util": "^5.0.0",
    "@config-plugins/react-native-pdf": "^5.0.0",
    "@expo/metro-config": "^0.10.7",
    "@expo/vector-icons": "^13.0.0",
    "@ptomasroos/react-native-multi-slider": "^2.2.2",
    "@react-native-async-storage/async-storage": "1.18.2",
    "@react-native-community/checkbox": "^0.5.14",
    "@react-native-community/datetimepicker": "7.2.0",
    "@react-native-community/slider": "4.4.2",
    "@react-native-firebase/analytics": "^18.5.0",
    "@react-native-firebase/app": "^18.5.0",
    "@react-native-firebase/auth": "^18.5.0",
    "@react-native-firebase/dynamic-links": "^18.5.0",
    "@react-native-firebase/firestore": "^18.5.0",
    "@react-native-firebase/functions": "^18.5.0",
    "@react-native-firebase/storage": "^18.5.0",
    "@react-native-picker/picker": "2.4.10",
    "@react-navigation/bottom-tabs": "^6.5.0",
    "@react-navigation/native": "^6.1.3",
    "@react-navigation/native-stack": "^6.9.5",
    "@redux-devtools/extension": "^3.2.5",
    "@reduxjs/toolkit": "^1.9.1",
    "@rneui/base": "^4.0.0-rc.7",
    "@rneui/themed": "^4.0.0-rc.7",
    "@types/react": "~18.2.14",
    "@typescript-eslint/parser": "^5.48.0",
    "crypto-js": "^4.1.1",
    "dotenv": "^16.0.3",
    "eslint": "^8.31.0",
    "eslint-config-airbnb": "^19.0.4",
    "expo": "^49.0.0",
    "expo-application": "~5.3.0",
    "expo-auth-session": "~5.0.2",
    "expo-av": "~13.4.1",
    "expo-background-fetch": "~11.3.0",
    "expo-blur": "~12.4.1",
    "expo-build-properties": "~0.8.3",
    "expo-calendar": "~11.3.2",
    "expo-checkbox": "~2.4.0",
    "expo-clipboard": "~4.3.1",
    "expo-constants": "~14.4.2",
    "expo-contacts": "~12.2.0",
    "expo-dev-client": "~2.4.11",
    "expo-device": "~5.4.0",
    "expo-document-picker": "~11.5.4",
    "expo-file-system": "~15.4.4",
    "expo-image-manipulator": "~11.3.0",
    "expo-image-picker": "~14.3.2",
    "expo-linear-gradient": "~12.3.0",
    "expo-linking": "~5.0.2",
    "expo-location": "~16.1.0",
    "expo-media-library": "~15.4.1",
    "expo-modules-core": "~1.5.10",
    "expo-notifications": "~0.20.1",
    "expo-random": "~13.2.0",
    "expo-sharing": "~11.5.0",
    "expo-sms": "~11.4.0",
    "expo-splash-screen": "~0.20.5",
    "expo-status-bar": "~1.6.0",
    "expo-updates": "~0.18.17",
    "expo-web-browser": "~12.3.2",
    "fbjs": "^3.0.5",
    "firebase": "^9.15.0",
    "firebase-admin": "^11.5.0",
    "firebase-functions": "^4.2.0",
    "gapi": "^0.0.3",
    "geolib": "^3.3.4",
    "i": "^0.3.7",
    "link-preview-js": "^3.0.5",
    "luxon": "^3.4.3",
    "moment": "^2.29.4",
    "npm": "^9.8.1",
    "postcode-validator": "^3.8.14",
    "radio-buttons-react-native": "^1.0.4",
    "react": "18.2.0",
    "react-content-loader": "^6.2.1",
    "react-dom": "18.2.0",
    "react-native": "0.72.6",
    "react-native-animated-pagination-dots": "^0.1.41",
    "react-native-blob-util": "^0.18.2",
    "react-native-cache": "^2.0.3",
    "react-native-calendars": "^1.1293.0",
    "react-native-collapsible": "^1.6.1",
    "react-native-compressor": "^1.6.1",
    "react-native-confetti-cannon": "^1.5.2",
    "react-native-deck-swiper": "^2.0.16",
    "react-native-element-dropdown": "^2.10.1",
    "react-native-emoji": "^1.8.0",
    "react-native-emoji-selector": "^0.2.0",
    "react-native-error-boundary": "^1.2.3",
    "react-native-gesture-handler": "~2.12.0",
    "react-native-image-picker": "^4.10.3",
    "react-native-image-viewing": "^0.2.2",
    "react-native-keyboard-aware-scroll-view": "^0.9.5",
    "react-native-markdown-display": "^7.0.2",
    "react-native-modal": "^13.0.1",
    "react-native-otp-auto-fill": "^0.1.17",
    "react-native-pager-view": "6.2.0",
    "react-native-paper": "^5.1.3",
    "react-native-pdf": "^6.6.2",
    "react-native-phone-number-input": "^2.1.0",
    "react-native-popup-menu": "^0.16.1",
    "react-native-progress": "^5.0.0",
    "react-native-recaptcha-that-works": "^1.3.2",
    "react-native-safe-area-context": "4.6.3",
    "react-native-screens": "~3.22.0",
    "react-native-section-list-get-item-layout": "^2.2.3",
    "react-native-share": "^8.2.2",
    "react-native-store-version": "^1.4.1",
    "react-native-svg": "13.9.0",
    "react-native-swipe-gestures": "^1.0.5",
    "react-native-swipe-list-view": "^3.2.9",
    "react-native-typing-animation": "^0.1.7",
    "react-native-vector-icons": "^9.2.0",
    "react-native-version-check-expo": "^3.4.7",
    "react-native-view-shot": "3.7.0",
    "react-native-webview": "13.2.2",
    "react-native-wheel-pick": "^1.2.0",
    "react-navigation": "^4.4.4",
    "react-redux": "^8.0.5",
    "react-redux-firebase": "^3.11.0",
    "redux": "^4.2.0",
    "redux-firestore": "^1.0.0",
    "rrule": "^2.7.1",
    "share-extension-expo-plugin": "^1.0.6",
    "validator": "^13.9.0",
    "zipcode": "^0.1.1"
  },
  "devDependencies": {
    "@babel/core": "^7.12.9",
    "@expo/webpack-config": "^19.0.0",
    "@types/react-native": "^0.70.8",
    "@typescript-eslint/eslint-plugin": "^5.48.0",
    "babel-plugin-module-resolver": "^5.0.0",
    "eslint-import-resolver-typescript": "^3.5.3",
    "eslint-plugin-import": "^2.26.0",
    "eslint-plugin-jsx-a11y": "^6.6.1",
    "eslint-plugin-react": "^7.31.11",
    "eslint-plugin-react-hooks": "^4.6.0",
    "react-native-svg-transformer": "^1.0.0",
    "typescript": "^5.1.3"
  },
}
```

#### `firebase.json` for react-native-firebase v6:

```json
# N/A
```

</details>

### iOS

<details><summary>Click To Expand</summary>
<p>

#### `ios/Podfile`:

- [X ] I'm not using Pods (Expo Managed Workflow)
- [] I'm using Pods and my Podfile looks like:
Including Pods install step from Expo in case it's helpful

```
RNFBAnalytics: Using default Firebase/Analytics with Ad Ids. May require App Tracking Transparency. Not allowed for Kids apps.
RNFBAnalytics: You may set variable `$RNFirebaseAnalyticsWithoutAdIdSupport=true` in Podfile to **e analytics without ad ids.
Using firebase.json from '/Users/expo/workingdir/build/firebase.json'
Adding a c**tom script phase for Pod RNFBApp: [RNFB] Core Configuration
Auto-linking React Native modules for target `ZealDev`: RNCAsyncStorage, RNCCheckbox, RNCPicker, RNDateTimePicker, RNFBAnalytics, RNFBApp, RNFBAuth, RNFBDynamicLinks, RNFBFirestore, RNFBFunctions, RNFBStorage, RNGestureHandler, RNSVG, RNScreens, RNShare, RNVectorIcons, amplitude-react-native, react-native-blob-util, react-native-compressor, react-native-image-picker, react-native-otp-auto-fill, react-native-pager-view, react-native-pdf, react-native-safe-area-context, react-native-slider, react-native-view-shot, and react-native-webview
Framework build type is static framework
[Codegen] Generating ./build/generated/ios/React-Codegen.podspec.json
Analyzing dependencies
Fetching podspec for `DoubleConversion` from `../node_modules/react-native/third-party-podspecs/DoubleConversion.podspec`
[Codegen] Found FBReactNativeSpec
Fetching podspec for `RCT-Folly` from `../node_modules/react-native/third-party-podspecs/RCT-Folly.podspec`
RNFBAnalytics: Using **er specified Firebase SDK version '10.22.0'
RNFBAnalytics: Using default Firebase/Analytics with Ad Ids. May require App Tracking Transparency. Not allowed for Kids apps.
RNFBAnalytics: You may set variable `$RNFirebaseAnalyticsWithoutAdIdSupport=true` in Podfile to **e analytics without ad ids.
RNFBApp: Using **er specified Firebase SDK version '10.22.0'
RNFBAuth: Using **er specified Firebase SDK version '10.22.0'
RNFBDynamicLinks: Using **er specified Firebase SDK version '10.22.0'
RNFBFirestore: Using **er specified Firebase SDK version '10.22.0'
RNFBFunctions: Using **er specified Firebase SDK version '10.22.0'
RNFBStorage: Using **er specified Firebase SDK version '10.22.0'
[Codegen] Found rncore
Fetching podspec for `boost` from `../node_modules/react-native/third-party-podspecs/boost.podspec`
Fetching podspec for `glog` from `../node_modules/react-native/third-party-podspecs/glog.podspec`
Adding spec repo `trunk` with CDN `https://cdn.cocoapods.org/`
CocoaPods 1.15.0 is available.
To update **e: `gem install cocoapods`
For more information, see https://blog.cocoapods.org and the CHANGELOG for this version at https://github.com/CocoaPods/CocoaPods/releases/tag/1.15.0
Downloading dependencies
Installing ASN1Decoder (1.10.0)
Installing BEMCheckBox (1.4.1)
Installing BoringSSL-GRPC (0.0.32)
Installing CometChatPro (3.0.910)
Installing DoubleConversion (1.1.6)
Installing EASClient (0.6.0)
Installing EXAV (13.4.1)
Installing EXApplication (5.3.1)
Installing EXBackgroundFetch (11.3.0)
Installing EXCalendar (11.3.2)
Installing EXConstants (14.4.2)
Installing EXContacts (12.2.0)
Installing EXFileSystem (15.4.4)
Installing EXFont (11.4.0)
Installing EXImageLoader (4.3.0)
Installing EXJSONUtils (0.7.1)
Installing EXLocation (16.1.0)
Installing EXManifests (0.7.2)
Installing EXMediaLibrary (15.4.1)
Installing EXNotifications (0.20.1)
Installing EXSplashScreen (0.20.5)
Installing EXStructuredHeaders (3.3.0)
Installing EXTaskManager (11.3.0)
Installing EXUpdates (0.18.17)
Installing EXUpdatesInterface (0.10.1)
Installing Expo (49.0.0)
Installing ExpoBlur (12.4.1)
Installing ExpoClipboard (4.3.1)
Installing ExpoCrypto (12.4.1)
Installing ExpoDevice (5.4.0)
Installing ExpoDocumentPicker (11.5.4)
Installing ExpoImageManipulator (11.3.0)
Installing ExpoImagePicker (14.3.2)
Installing ExpoKeepAwake (12.3.0)
Installing ExpoLinearGradient (12.3.0)
Installing ExpoLocalization (14.1.1)
Installing ExpoModulesCore (1.5.11)
Installing ExpoRandom (13.2.0)
Installing ExpoSMS (11.4.0)
Installing ExpoSharing (11.5.0)
Installing ExpoWebBrowser (12.3.2)
Installing FBLazyVector (0.72.6)
Installing FBReactNativeSpec (0.72.6)
Installing Firebase (10.22.0)
Installing FirebaseAnalytics (10.22.0)
Installing FirebaseAppCheckInterop (10.23.0)
Installing FirebaseAuth (10.22.0)
Installing FirebaseAuthInterop (10.23.0)
Installing FirebaseCore (10.22.0)
Installing FirebaseCoreExtension (10.23.0)
Installing FirebaseCoreInternal (10.23.0)
Installing FirebaseDynamicLinks (10.22.0)
Installing FirebaseFirestore (10.22.0)
Installing FirebaseFirestoreInternal (10.23.0)
Installing FirebaseFunctions (10.22.0)
Installing FirebaseInstallations (10.23.0)
Installing FirebaseMessagingInterop (10.23.0)
Installing FirebaseSharedSwift (10.23.0)
Installing FirebaseStorage (10.22.0)
Installing GTMSessionFetcher (3.3.2)
Installing GoogleAppMeasurement (10.22.0)
Installing GoogleUtilities (7.13.0)
Installing NextLevelSessionExporter (0.4.6)
Installing PromisesObjC (2.4.0)
Installing RCT-Folly (2021.07.22.00)
Installing RCTRequired (0.72.6)
Installing RCTTypeSafety (0.72.6)
Installing RNCAsyncStorage (1.18.2)
Installing RNCCheckbox (0.5.15)
Installing RNCPicker (2.4.10)
Installing RNDateTimePicker (7.2.0)
Installing RNFBAnalytics (18.5.0)
Installing RNFBApp (18.5.0)
Installing RNFBAuth (18.0.0)
Installing RNFBDynamicLinks (18.0.0)
Installing RNFBFirestore (18.0.0)
Installing RNFBFunctions (18.0.0)
Installing RNFBStorage (18.1.0)
Installing RNGestureHandler (2.12.1)
Installing RNSVG (13.9.0)
Installing RNScreens (3.22.1)
Installing RNShare (8.2.2)
Installing RNVectorIcons (9.2.0)
Installing ReachabilitySwift (5.2.1)
Installing React (0.72.6)
Installing React-Codegen (0.72.6)
Installing React-Core (0.72.6)
Installing React-CoreModules (0.72.6)
Installing React-NativeModulesApple (0.72.6)
Installing React-RCTActionSheet (0.72.6)
Installing React-RCTAnimation (0.72.6)
Installing React-RCTAppDelegate (0.72.6)
Installing React-RCTBlob (0.72.6)
Installing React-RCTImage (0.72.6)
Installing React-RCTLinking (0.72.6)
Installing React-RCTNetwork (0.72.6)
Installing React-RCTSettings (0.72.6)
Installing React-RCTText (0.72.6)
Installing React-RCTVibration (0.72.6)
Installing React-callinvoker (0.72.6)
Installing React-cxxreact (0.72.6)
Installing React-debug (0.72.6)
Installing React-jsc (0.72.6)
Installing React-jsi (0.72.6)
Installing React-jsiexecutor (0.72.6)
Installing React-jsinspector (0.72.6)
Installing React-logger (0.72.6)
Installing React-perflogger (0.72.6)
Installing React-rncore (0.72.6)
Installing React-runtimeexecutor (0.72.6)
Installing React-runtimescheduler (0.72.6)
Installing React-utils (0.72.6)
Installing ReactCommon (0.72.6)
Installing RecaptchaInterop (100.0.0)
Installing ShareExtensionNative (1.0.6)
Installing SocketRocket (0.6.1)
Installing UMAppLoader (4.2.0)
Installing Yoga (1.14.0)
Installing abseil (1.20240116.1)
Installing amplitude-react-native (1.4.5)
Installing boost (1.76.0)
Installing expo-dev-client (2.4.11)
Installing expo-dev-launcher (2.4.13)
Installing expo-dev-menu (3.2.1)
Installing expo-dev-menu-interface (1.3.0)
Installing fmt (6.2.1)
Installing gRPC-C++ (1.62.1)
Installing gRPC-Core (1.62.1)
Installing glog (0.3.5)
Installing leveldb-library (1.22.4)
Installing nanopb (2.30909.1)
Installing react-native-blob-util (0.18.3)
Installing react-native-compressor (1.6.1)
Installing react-native-image-picker (4.10.3)
Installing react-native-otp-auto-fill (0.1.17)
Installing react-native-pager-view (6.2.0)
Installing react-native-pdf (6.6.2)
Installing react-native-safe-area-context (4.6.3)
Installing react-native-slider (4.4.2)
Installing react-native-view-shot (3.7.0)
Installing react-native-webview (13.2.2)
Generating Pods project
```

</p>
</details>

---

## Environment

<details><summary>Click To Expand</summary>
<p>
Ran `npx expo-env-info` since react-native-info was not working

**`expo-env-info` output:**

```
  expo-env-info 1.2.0 environment info:
    System:
      OS: macOS 13.4
      Shell: 3.2.57 - /bin/sh
    Binaries:
      Node: 18.17.1 - /usr/local/bin/node
      npm: 9.8.1 - ~/Projects/Zeal/joyapp/node_modules/.bin/npm
      Watchman: 2024.01.22.00 - /opt/homebrew/bin/watchman
    SDKs:
      iOS SDK:
        Platforms: DriverKit 22.4, iOS 16.4, macOS 13.3, tvOS 16.4, watchOS 9.4
    IDEs:
      Xcode: 14.3.1/14E300c - /usr/bin/xcodebuild
    npmPackages:
      @expo/metro-config: ^0.10.7 => 0.10.7 
      @expo/webpack-config: ^19.0.0 => 19.0.0 
      expo: ^49.0.0 => 49.0.0 
      react: 18.2.0 => 18.2.0 
      react-dom: 18.2.0 => 18.2.0 
      react-native: 0.72.6 => 0.72.6 
      react-navigation: ^4.4.4 => 4.4.4 
    npmGlobalPackages:
      eas-cli: 7.5.0
      expo-cli: 5.4.0
    Expo Workflow: managed
```

- **Platform that you're experiencing the issue on**:
  - [X ] iOS
  - [ ] Android
  - [X ] **iOS** but have not tested behavior on Android
  - [ ] **Android** but have not tested behavior on iOS
  - [ ] Both
- **`react-native-firebase` version you're using that has this issue:**
  - 18.5.0
- **`Firebase` module(s) you're using that has the issue:**
  - "@react-native-firebase/analytics": "^18.5.0",
    "@react-native-firebase/app": "^18.5.0",
    "@react-native-firebase/auth": "^18.0.0",
    "@react-native-firebase/dynamic-links": "^18.0.0",
    "@react-native-firebase/firestore": "^18.0.0",
    "@react-native-firebase/functions": "^18.0.0",
    "@react-native-firebase/storage": "^18.1.0",
- **Are you using `TypeScript`?**
  - Y 5.2.2

</p>
</details>

---

- 👉 Check out [`React Native Firebase`](https://twitter.com/rnfirebase) and [`Invertase`](https://twitter.com/invertaseio) on Twitter for updates on the library.

**Comment by jakechao-zeal** [comment]
Adding a note here that I tried upgrading to Expo SDK 50 and the issue seems to have gone away. I am not sure if I am ready yet to upgrade to Expo SDK 50 due to other issues I've seen that upgrade cause, so I'm leaving this issue open in case anyone can provide guidance as to why my environment and dependencies using Expo SDK 49 (listed in original issue comment) would be experiencing this issue? 


Seems like the only relevant package update would be react native? 

New `package.json` after Expo SDK 50 upgrade

```
  @expo/metro-config@0.10.7 - expected version: ~0.17.1
  @expo/vector-icons@13.0.0 - expected version: ^14.0.0
  @react-native-async-storage/async-storage@1.18.2 - expected version: 1.21.0
  @react-native-community/datetimepicker@7.2.0 - expected version: 7.6.1
  @react-native-picker/picker@2.4.10 - expected version: 2.6.1
  @types/react@18.2.33 - expected version: ~18.2.45
  expo-application@5.3.1 - expected version: ~5.8.3
  expo-auth-session@5.0.2 - expected version: ~5.4.0
  expo-av@13.4.1 - expected version: ~13.10.5
  expo-background-fetch@11.3.0 - expected version: ~11.8.1
  expo-blur@12.4.1 - expected version: ~12.9.2
  expo-build-properties@0.8.3 - expected version: ~0.11.1
  expo-calendar@11.3.2 - expected version: ~12.2.1
  expo-checkbox@2.4.0 - expected version: ~2.7.0
  expo-clipboard@4.3.1 - expected version: ~5.0.1
  expo-constants@14.4.2 - expected version: ~15.4.5
  expo-contacts@12.2.0 - expected version: ~12.8.2
  expo-dev-client@2.4.13 - expected version: ~3.3.11
  expo-device@5.4.0 - expected version: ~5.9.3
  expo-document-picker@11.5.4 - expected version: ~11.10.1
  expo-file-system@15.4.5 - expected version: ~16.0.8
  expo-image-manipulator@11.3.0 - expected version: ~11.8.0
  expo-image-picker@14.3.2 - expected version: ~14.7.1
  expo-linear-gradient@12.3.0 - expected version: ~12.7.2
  expo-linking@5.0.2 - expected version: ~6.2.2
  expo-location@16.1.0 - expected version: ~16.5.5
  expo-media-library@15.4.1 - expected version: ~15.9.1
  expo-notifications@0.20.1 - expected version: ~0.27.6
  expo-random@13.2.0 - expected version: ~13.6.0
  expo-sharing@11.5.0 - expected version: ~11.10.0
  expo-sms@11.4.0 - expected version: ~11.7.1
  expo-splash-screen@0.20.5 - expected version: ~0.26.4
  expo-status-bar@1.6.0 - expected version: ~1.11.1
  expo-updates@0.18.19 - expected version: ~0.24.12
  expo-web-browser@12.3.2 - expected version: ~12.8.2
  react-native@0.72.10 - expected version: 0.73.6
  react-native-gesture-handler@2.12.1 - expected version: ~2.14.0
  react-native-pager-view@6.2.0 - expected version: 6.2.3
  react-native-safe-area-context@4.6.3 - expected version: 4.8.2
  react-native-screens@3.22.1 - expected version: ~3.29.0
  react-native-svg@13.9.0 - expected version: 14.1.0
  react-native-view-shot@3.7.0 - expected version: 3.8.0
  react-native-webview@13.2.2 - expected version: 13.6.4
  @expo/webpack-config@19.0.0 - expected version: ~19.0.1
  typescript@5.2.2 - expected version: ^5.3.0
```

New environment after Expo SDK 50 upgrade

```
expo-env-info 1.2.0 environment info:
    System:
      OS: macOS 13.4
      Shell: 3.2.57 - /bin/sh
    Binaries:
      Node: 18.17.1 - /usr/local/bin/node
      npm: 9.6.7 - /usr/local/bin/npm
      Watchman: 2024.01.22.00 - /opt/homebrew/bin/watchman
    SDKs:
      iOS SDK:
        Platforms: DriverKit 22.4, iOS 16.4, macOS 13.3, tvOS 16.4, watchOS 9.4
    IDEs:
      Xcode: 14.3.1/14E300c - /usr/bin/xcodebuild
    npmPackages:
      @expo/metro-config: ~0.17.1 => 0.17.6 
      @expo/webpack-config: ~19.0.1 => 19.0.1 
      expo: ^50.0.4 => 50.0.14 
      react: 18.2.0 => 18.2.0 
      react-dom: 18.2.0 => 18.2.0 
      react-native: 0.73.6 => 0.73.6 
      react-navigation: ^4.4.4 => 4.4.4 
    npmGlobalPackages:
      eas-cli: 7.5.0
      expo-cli: 5.4.0
    Expo Workflow: managed
```

**Comment by jillhc** [comment]
Seeing exactly the same crash in our project, that uses RN Firebase + Expo. 

A couple notes to add:
- This crash only happens on the archived build (archived using Expo and distributed via TestFlight). It does not occur when running locally on device. We tried turning on code optimization and running expo in production mode `npx expo start --no-dev --minify` , but have been unable to reproduce the crash with those methods.

- This crash happens upon login or account creation. The original reported mentioned crashing "on startup". I was able to downgrade the build, log out of the Firebase auth account, and then upgrade the build. In the logged out state, the app runs fine! It's only when authenticating that it crashes. 

- Archiving on my computer using Xcode produces as crash-free app! However, it's way more preferable to use Expo for the build process.


@jakechao-zeal have you reported to Expo too? I can share these notes there as well.
